src/main/java/utils/CustomFileReaderFactory.java

```diff
@@ -0,0 +1,21 @@
+package utils;
+
+import utils.filereaders.CsvFileReader;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CustomFileReaderFactory {
+    private static final HashMap<FileType, CustomFileReader> CUSTOM_FILE_READER = new HashMap<>(Map.of(
```

Comment:
replace this with ImmutableMap, use the builder from it, it comes from guava i think, or google commons